### ISSUE TITLE
onOpen trigger onCheckAll?

### ISSUE THREAD

**Issue by jovvybersamin** [wenzhixin/multiple-select#296]
I'm using the latest version once I tried to open the multiple select.

the onOpen callback also calls the onCheckAll? This is a big problem to me because my onCheckAll callback sent an ajax request to my server. Therefore the option lists element are not showing.

**Comment by slavede** [comment]
Noticed the same...

**Comment by themightystephen** [comment]
Same issue here.

**Comment by wenzhixin** [comment]
Fixed in the latest version: http://multiple-select.wenzhixin.net.cn/examples/#events.html

**Comment by themightystephen** [comment]
Hi @wenzhixin, I've pulled the latest code (v1.3.0) but still seem to be facing the same issue. The `onCheckAll` callback appears to still be called when opening the multiple select. I've added a `console.trace()` to my `onCheckAll` event handler to trace the cause:

**dashboard.js**
```js
$('select#view-as').multipleSelect({
        placeholder: "View As",
        filter: true,

        selectAll: true, // if I set this to false, the issue doesn't happen (but unfortunately I lose "Select All" functionality too)
        allSelected: "All Users",
        width: 400,
        multiple: true,
        multipleWidth: 180,
        minimumCountSelected: 4,

        onClick: function(view) {
            updateViewers(view);
            refreshAllReports();
        },
        onOptgroupClick: function(view) {
            updateViewers(view);
            refreshAllReports();
        },
        onCheckAll: function() {
           console.trace(); // line 1924
           updateViewers(view);
            refreshAllReports();
        },
        onUncheckAll: function() {
            updateViewers(view);
            refreshAllReports();
        }
    });
```

The following stack trace appears when I open the multiple select:

  | onCheckAll | @ | dashboard.js:1924
-- | -- | -- | --
  | updateSelectAll | @ | multiple-select.js:1098
  | filter | @ | multiple-select.js:1277
  | open | @ | multiple-select.js:1014
  | toggleOpen | @ | multiple-select.js:865
  | dispatch | @ | jquery.js:4435
  | r.handle | @ | jquery.js:4121

From the stack trace it appears `onCheckAll` is being called by `open` via `filter`.

NOTE: I'm using JQuery v2.1.4.

**Comment by wenzhixin** [comment]
OK, I will test it.

**Comment by wenzhixin** [comment]
@themightystephen try this jsfiddle:  http://jsfiddle.net/wenyi/knbdm5hu/2344/

**Comment by wenzhixin** [comment]
Fixed in #454

**Comment by themightystephen** [comment]
Thanks @wenzhixin! I've tested out your dev branch with my project and can confirm this is resolved. Thank you again. Is there an expected date when this will be included in an official release?

**Comment by wenzhixin** [comment]
https://github.com/wenzhixin/multiple-select/releases
Released v1.4.0 2 days ago.

**Comment by themightystephen** [comment]
Ah sorry, things have changed since when I checked on Friday. Fantastic!
